diff --git a/m4/curl-functions.m4 b/m4/curl-functions.m4
index 425d9f96f8..da7131db58 100644
--- a/m4/curl-functions.m4
+++ b/m4/curl-functions.m4
@@ -3592,7 +3592,7 @@ AC_DEFUN([CURL_CHECK_FUNC_IOCTLSOCKET_FIONBIO], [
       AC_LANG_PROGRAM([[
         $curl_includes_winsock2
       ]],[[
-        int flags = 0;
+        unsigned long flags = 0;
         if(0 != ioctlsocket(0, FIONBIO, &flags))
           return 1;
       ]])
