diff --git a/Makefile b/Makefile
index b361a63..039539d 100644
--- a/Makefile
+++ b/Makefile
@@ -16,7 +16,7 @@ else
 OPTCFLAGS += -Ofast -ffast-math
 endif
 # Use the 1999 ISO C standard with POSIX.1-2008 definitions.
-CFLAGS = -std=c99 -D_POSIX_C_SOURCE=200809L -Wall -Wno-maybe-uninitialized -pipe -I. $(OPTCFLAGS)
+CFLAGS = -std=c99 -D_POSIX_C_SOURCE=200809L -pipe -I. $(OPTCFLAGS)
 
 ifeq ($(LIBRARY_CONFIGURATION), SHARED)
 # Shared library.
diff --git a/misc.c b/misc.c
index 52d339f..373f02d 100644
--- a/misc.c
+++ b/misc.c
@@ -81,7 +81,7 @@ void detexSetErrorMessage(const char *format, ...) {
 	va_start(args, format);
 	char *message;
 	// Allocate and set message.
-	int r = vasprintf(&message, format, args);
+	int r = 0;
 	if (r < 0)
 		message = strdup("detexSetErrorMessage: vasprintf returned error");
 	va_end(args);
