diff --git a/Makethird b/Makethird
index 68b4b694c..7e4212d97 100644
--- a/Makethird
+++ b/Makethird
@@ -199,7 +199,22 @@ THIRD_SRC += thirdparty/libjpeg/jmemmgr.c
 THIRD_SRC += thirdparty/libjpeg/jquant1.c
 THIRD_SRC += thirdparty/libjpeg/jquant2.c
 THIRD_SRC += thirdparty/libjpeg/jutils.c
-
+THIRD_SRC += thirdparty/libjpeg/jcapimin.c
+THIRD_SRC += thirdparty/libjpeg/jcapistd.c
+THIRD_SRC += thirdparty/libjpeg/jcmarker.c
+THIRD_SRC += thirdparty/libjpeg/jcparam.c
+THIRD_SRC += thirdparty/libjpeg/jctrans.c
+THIRD_SRC += thirdparty/libjpeg/jcinit.c
+THIRD_SRC += thirdparty/libjpeg/jcarith.c
+THIRD_SRC += thirdparty/libjpeg/jccoefct.c
+THIRD_SRC += thirdparty/libjpeg/jccolor.c
+THIRD_SRC += thirdparty/libjpeg/jcdctmgr.c
+THIRD_SRC += thirdparty/libjpeg/jchuff.c
+THIRD_SRC += thirdparty/libjpeg/jcmainct.c
+THIRD_SRC += thirdparty/libjpeg/jcmaster.c
+THIRD_SRC += thirdparty/libjpeg/jcprepct.c
+THIRD_SRC += thirdparty/libjpeg/jcsample.c
+THIRD_SRC += thirdparty/libjpeg/jmemansi.c
 THIRD_CFLAGS += -Iscripts/libjpeg -Ithirdparty/libjpeg
 
 $(OUT)/thirdparty/libjpeg/%.o: thirdparty/libjpeg/%.c
diff --git a/include/mupdf/fitz/config.h b/include/mupdf/fitz/config.h
index ea91e876d..ce3c26b48 100644
--- a/include/mupdf/fitz/config.h
+++ b/include/mupdf/fitz/config.h
@@ -19,6 +19,12 @@
 /* #define FZ_PLOTTERS_CMYK 1 */
 /* #define FZ_PLOTTERS_N 1 */
 
+#define FZ_PLOTTERS_G 0
+#define FZ_PLOTTERS_RGB 1
+#define FZ_PLOTTERS_CMYK 0
+#define FZ_PLOTTERS_N 0
+
+
 /*
 	Choose which document agents to include.
 	By default all are enabled. To avoid building unwanted
@@ -33,9 +39,18 @@
 /* #define FZ_ENABLE_EPUB 1 */
 
 /*
++#define FZ_ENABLE_PDF 1
++#define FZ_ENABLE_XPS 0
++#define FZ_ENABLE_SVG 1
++#define FZ_ENABLE_CBZ 0
++#define FZ_ENABLE_IMG 1
++#define FZ_ENABLE_HTML 0
++#define FZ_ENABLE_EPUB 0
+
+
 	Choose whether to enable ICC color profiles.
 */
-/* #define FZ_ENABLE_ICC 1 */
+#define FZ_ENABLE_ICC 1
 
 /*
 	Choose whether to enable JPEG2000 decoding.
@@ -50,6 +65,7 @@
 	By default JavaScript is enabled both for mutool and PDF interactivity.
 */
 /* #define FZ_ENABLE_JS 1 */
+#define FZ_ENABLE_JS 0
 
 /*
 	Choose which fonts to include.
@@ -60,22 +76,22 @@
 	unwanted fonts.
 */
 /* To avoid all noto fonts except CJK, enable: */
-/* #define TOFU */
+#define TOFU
 
 /* To skip the CJK font, enable: (this implicitly enables TOFU_CJK_EXT and TOFU_CJK_LANG) */
-/* #define TOFU_CJK */
+#define TOFU_CJK
 
 /* To skip CJK Extension A, enable: (this implicitly enables TOFU_CJK_LANG) */
-/* #define TOFU_CJK_EXT */
+#define TOFU_CJK_EXT
 
 /* To skip CJK language specific fonts, enable: */
-/* #define TOFU_CJK_LANG */
+#define TOFU_CJK_LANG
 
 /* To skip the Emoji font, enable: */
 /* #define TOFU_EMOJI */
 
 /* To skip the ancient/historic scripts, enable: */
-/* #define TOFU_HISTORIC */
+#define TOFU_HISTORIC
 
 /* To skip the symbol font, enable: */
 /* #define TOFU_SYMBOL */
